src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1437,11 +1457,60 @@ internal static int IndexOfNewlineChar(ReadOnlySpan<char> text, out int stride)
                     if ((uint)nextCharIdx < (uint)text.Length && text[nextCharIdx] == '\n')
                     {
                         stride = 2;
+
+                        if (replacementText != "\r\n")
+                        {
+                            return offset;
+                        }
                     }
+                    else if (replacementText != "\r")
+                    {
+                        return offset;
+                    }
+                }
+                else if (replacementText.Length != 1 || replacementText[0] != text[idx])
+                {
+                    return offset;
                 }
+
+                offset += stride;
+                text = text.Slice(idx + stride);
+            }
+        }
+
+        private string ReplaceLineEndingsWithLineFeed()
+        {
+            // If we are going to replace the new line with a line feed ('\n'),
+            // we can skip looking for it to avoid breaking out of the vectorized path unnecessarily.
+            // We can't use the same optimization for the carriage return ('\r')
+            // as we still want to replace CRLF ("\r\n") sequences.
```

Comment:
```c#
int stride = 1;
if (this[idxOfFirstNewlineChar] == '\n' &&
   idxOfFirstNewlineChar > 0 &&
   this[idxOfFirstNewlineChar - 1] == '\r')
{
    idxOfFirstNewlineChar--;
    stride=2;
}
```
would this not work and have the similar performance when replacing with carriage return?

Reply:
It should be possible -- I'll update the comment.
I don't think it's worth the extra code to implement that though as I've never seen `ReplaceLineEndings("\r")` being used.